test/integration/fiscalYear.extra.js

```diff
@@ -0,0 +1,144 @@
+   * from the date 2017-08-24
+   */
+  it(`GET /fiscal/date?date=${flatDate(date)} returns the fiscal year from a given date`, () => {
+    return agent.get(url.concat(`/date?date=${date}`))
+      .then(res => {
+        helpers.api.listed(res, 1);
+        const [fy] = res.body;
+        expect(fy).to.be.an('object');
+        expect(fy).to.have.property('fiscal_year_id', 3);
+        expect(fy).to.have.property('previous_fiscal_year_id', 2);
+        expect(fy).to.have.property('label', 'Test Fiscal Year 2017');
+
+        /**
+         * dates are returned with timezone information
+         * we extract just the date information
+         * FIXME: timezone issue : we have 2017-12-31 in the database
+         * but we get 2018-01-01, it is now difficult to test date.
+         */
+        expect(fy).to.have.property('start_date');
+        expect(fy).to.have.property('end_date');
+        const fyStartDate = flatDate(fy.start_date);
+        const fyEndDate = flatDate(fy.end_date);
+        expect(fyStartDate).to.be.equal('2017-01-01');
+        expect(fyEndDate).to.be.equal('2017-12-31');
+
+        /**
+         * the percentage is value of consumed period in the fiscal year
+         * until to the given date
+         * proportion = (currentDate - startDate) / (nbMonthOfYear * 30.5)
+         * proportion = ('2017-08-24' - '2017-01-01') / (12 * 30.5)
+         * proportion = 235 / 366 = 0.6421 = 64.21%
+         */
+        expect(fy).to.have.property('percentage', 0.6421);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('POST /fiscal/:id/opening_balance forbid to set the opening balance of a fiscal year which is not the first', () => {
+    // define a not first fiscal year information
+    openingBalance.id = fiscalYear2017.id;
+    openingBalance.fiscal = fiscalYear2017;
+    openingBalance.accounts = accounts;
+
+    return agent.post(url.concat(`/${fiscalYear2017.id}/opening_balance`))
+      .send({ params : openingBalance })
+      .then(res => {
+        helpers.api.errored(res, 400, 'ERRORS.NOT_BEGINING_FISCAL_YEAR');
+      })
+      .catch(helpers.handler);
+  });
+
+  it('POST /fiscal/:id/opening_balance set the opening balance of a fiscal year', () => {
+    // define a not first fiscal year information
+    openingBalance.id = fiscalYear2015.id;
+    openingBalance.fiscal = fiscalYear2015;
+    openingBalance.accounts = accounts;
+
+    return agent.post(url.concat(`/${fiscalYear2015.id}/opening_balance`))
+      .send({ params : openingBalance })
+      .then(res => {
+        expect(res).to.have.status(201);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /fiscal/:id/opening_balance returns the opening balance of a given fiscal year', () => {
+    return agent.get(url.concat(`/${fiscalYear2015.id}/opening_balance`))
+      .then(res => {
+        const updatedAccounts = res.body.filter(item => {
+          return [81, 83, 174, 190].indexOf(item.id) > -1;
```

Comment:
You can also use `includes()`!  It would make it more readable to name the array inside as well.  Like this:

```js
const accountSubset = [81, 83, 174, 190];
const updatedAccounts = res.body.filter(account => accountSubset.includes(account.id));
```

Reply:
It might be a good idea to sort it as well after downloading it - so that any changes in ordering do not break your test.

```js
updatedAccounts.sort();
// then
const [first, second, third, fourth] = updatedAccounts;
```

Reply:
You're right 👍